astropy/wcs/tests/test_wcs.py

```diff
@@ -954,47 +954,48 @@ def test_passing_ImageHDU():
     assert wcs_hdu.wcs.compare(wcs_header.wcs)
 
 
-def inconsistent_sip():
+def test_inconsistent_sip():
     """
     Test for #4814
     """
     hdr = get_pkg_data_contents("data/sip-broken.hdr")
     w = wcs.WCS(hdr)
     newhdr = w.to_header(relax=None)
     # CTYPE should not include "-SIP" if relax is None
-    assert(all([ctyp[-4 :] != '-SIP' for ctyp in self.wcs.ctype]))
+    wnew = wcs.WCS(newhdr)
+    assert(all([ctyp[-4 :] != '-SIP' for ctyp in wnew.wcs.ctype]))
```

Comment:
The `ctype[-4 :] != '-SIP'` could be written as `not ctype.endswith('-SIP')` and the list comprehension could be omitted as well as the paranthesis after `assert`:

`assert all(not ctype.endswith('-SIP') for ctyp in wnew.wcs.ctype)` or `assert not any(ctype.endswith('-SIP') for ctyp in wnew.wcs.ctype)`.

However, that's not really essential so feel free to ignore this comment.